app/src/ui/banners/rebase-conflicts-banner.tsx

```diff
@@ -20,16 +20,22 @@ export class RebaseConflictsBanner extends React.Component<
 > {
   private openDialog = () => {
     this.props.onDismissed()
-    this.props.dispatcher.showPopup(this.props.popup)
+    this.props.onOpenDialog()
     this.props.dispatcher.recordRebaseConflictsDialogReopened()
   }
 
+  private onDismissed = () => {
+    log.warn(
+      `[RebaseConflictsBanner] this is not dismissable by default unless the user clicks on the link`
```

Comment:
I'm not a huge fan of this callback because when I've set `dismissable` to `false` (expecting that this component will handle it's dismissal) it then feels unnecessary to require a `onDismissed` prop, but maybe I'm using this wrong. Thoughts?

Reply:
hmmm yeah that's interesting. i think my thought around requiring the `onDismissed` prop was that the banner would always dismiss itself, but i think the is a good counter example to that. i'd be happy with making the `onDismissed` prop in the banner component optional so you don't have to pass this in. how does that sit with you?

Reply:
I think this needs a bigger discussion than this change because `Dialog` is setup the same way, and making `onDismissed` optional would just move the problem elsewhere:

 - a `Banner` could somehow be set to `dismissable={true}` and `onDismissed={undefined}` due to a coding error
 - user click `Close` to dismiss the banner
 - what should happen here when `onDismissed` is undefined - a runtime error? a `development` runtime error but is silently ignored in `production`? something else?

https://github.com/desktop/desktop/blob/2d30990632997cbc192e7cc03ddae40aa819a4f4/app/src/ui/banners/banner.tsx#L39-L43

<img width="658" alt="Screen Shot 2019-03-25 at 2 07 30 PM" src="https://user-images.githubusercontent.com/359239/54939496-823e5800-4f07-11e9-805c-3b34ccf6b3fa.png">

 I'm happy to leave this warning in here for the moment rathe than change the accepted props values, but I'd like to think more about alternative ways to ensure the component doesn't get put in an invalid state.

Reply:
okay! that makes sense!